packages/boot/src/mixins/boot.mixin.ts

```diff
@@ -189,14 +196,27 @@ export function BootMixin<T extends MixinTarget<Application>>(superClass: T) {
      *
      * @param component - The component to mount booters of
      */
-    mountComponentBooters(component: Constructor<{}>) {
-      const componentKey = `components.${component.name}`;
-      const compInstance = this.getSync<{
-        booters?: Constructor<Booter>[];
-      }>(componentKey);
-
-      if (compInstance.booters) {
-        this.booters(...compInstance.booters);
+    mountComponentBooters(
+      componentInstanceOrClass: Constructor<unknown> | InstanceWithBooters,
+    ) {
+      const componentInstance = resolveComponentInstance(this);
+      if (componentInstance.booters) {
+        this.booters(...componentInstance.booters);
+      }
+
+      /**
+       * Determines if componentInstanceOrClass is an instance of a component,
+       * or a class that needs to be instantiated from context.
+       * @param ctx
+       */
+      function resolveComponentInstance(ctx: Readonly<Context>) {
+        if (typeof componentInstanceOrClass !== 'function') {
+          return componentInstanceOrClass;
+        }
+
+        const componentName = componentInstanceOrClass.name;
```

Comment:
Please add a `TODO(semver-major)` comment to remind us to remove this code when working on the next major release.